src/Flow.jl

```diff
@@ -60,20 +59,15 @@ lowerBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop (
 upperBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop r[I-δ(j,I),i] -= Φ[CIj(j,I,2)] over I ∈ slice(N,N[j],j,2)
 
 """
-    applyBodyForce!(r,t,g=(a,b)->0,Val{Bool})
+    accelerate!(r,t,g)
 
-This function apply body force `g` at time `t` to `r`, effectively the du/dt.
-Body force `g` input sould be function with arguments: index `a` and time `b`.
-The default of body force is null. Val{Bool} is applied to specify whether to run or not.
+This function apply uniform acceleration field `g` at time `t` to `r`.
+If `g ≠ nothing`, then `g(i,t)=dUᵢ/dt`.
 """
-function applyBodyForce!(r,t,::Val{true},g=(a,b)->0)
-    N,n = size_u(r)
-    for i ∈ 1:n
-        r[CartesianIndices(N),i] .+= g(i,t)
-    end
+accelerate!(r,t,g) = for i ∈ 1:last(size(r))
+    r[:,i] .= g(i,t)
```

Comment:
I think `r[CartesianIndices(N),i] .+= g(i,t)` is still necessary since the colon of `r[:,i]` can only get slice for one dimension, which cannot handle if dim(r)>2. 

Also I am wandering why the forcing vector `r` is overwritten. I guess it should be augmented. That is, `.=` -> `.+=`
Because in multiphase flow I have to 1) a.f = \mu(gradU+gradU^T), 2) a.f ./= rho(f), 3) a.f .+= convection term 4) add gravity.
No matter the order of 3) or 4), we need to augment the body force term after finishing the calculation of diffusion term. step 1) and 2) cannot be combined together.

Reply:
> I think `r[CartesianIndices(N),i]` is still necessary since the colon of `r[:,i]` can only get slice for one dimension, which cannot handle if dim(r)>2. (runtests.jl will yield the error)
Oops. I'm not sure how that didn't error when I checked it, but you're right.

> Also I am wandering why the forcing vector `r` is overwritten. I guess it should be augmented. That is, `.=` -> `.+=` Because in multiphase flow I have to 1) a.f = \mu(gradU+gradU^T), 2) a.f ./= rho(f), 3) a.f .+= convection term 4) add gravity. No matter the order of 3) or 4), we need to augment the body force term after finishing the calculation of diffusion term. step 1) and 2) cannot be combined together.

See line 100 of main.f90 in my old Fortran solver. The first step of every loop is setting up the background acceleration field.

Reply:
Besides, the multi-phase stuff will need to be a whole different type, right? Not `Flow`, but `TwoPhase` or something. It'll need it's own `mom_step` function, etc. So there is no need to make this PR match exactly to what you're going to do in that function.

Reply:
But, I'm happy going back to the `.+=` version. I'll push that up.

Reply:
The reason why in the fortran code we could specify domain acceleration first is because you have defined another variables `v` in line 79 in fvel.f90, However, in order to save the memory, we did not have any spare vector variable. as a result we need to divide `a.f` in situ when it comes to diffusion term. I appreciate for your understanding.